src/platform/modulesmanager/src/ManagementModule.cpp

```diff
@@ -471,14 +471,179 @@ bool ManagementModule::IsExportingMmi(const std::string path)
+                    "anyOf": [
+                        {
+                            "$ref": "#/definitions/string"
+                        },
+                        {
+                            "$ref": "#/definitions/integer"
+                        },
+                        {
+                            "$ref": "#/definitions/boolean"
+                        },
+                        {
+                            "$ref": "#/definitions/integerEnumeration"
+                        },
+                        {
+                            "$ref": "#/definitions/stringArray"
+                        },
+                        {
+                            "$ref": "#/definitions/integerArray"
+                        },
+                        {
+                            "$ref": "#/definitions/stringMap"
+                        },
+                        {
+                            "$ref": "#/definitions/integerMap"
+                        }
+                    ]
+                }
+            },
+            "objectArray": {
+                "type": "array",
+                "items": {
+                    "$ref": "#/definitions/object"
+                }
+            }
+        },
+        "anyOf": [
+            {
+                "$ref": "#/definitions/string"
+            },
+            {
+                "$ref": "#/definitions/integer"
+            },
+            {
+                "$ref": "#/definitions/boolean"
+            },
+            {
+                "$ref": "#/definitions/object"
+            },
+            {
+                "$ref": "#/definitions/objectArray"
+            }
+        ]
+    })""";
+
+    rapidjson::Document sd;
+    sd.Parse(schemaJson);
+    rapidjson::SchemaDocument schema(sd);
+
+    std::string payloadString = std::string(payload, payloadSizeBytes);
+    OsConfigLogInfo(ModulesManagerLog::Get(), "%s", payloadString.c_str());
+
+    rapidjson::Document document;
+    if (document.Parse(payload, payloadSizeBytes).HasParseError()) {
+        OsConfigLogError(ModulesManagerLog::Get(), "Invalid JSON payload");
+        isValid = false;
+    }
+    else
+    {
+        rapidjson::SchemaValidator validator(schema);
+        if (!document.Accept(validator))
+        {
+            rapidjson::StringBuffer sb;
+            std::string invalidDocumentPointer;
+            std::string invalidSchemaPointer;
+
+            validator.GetInvalidDocumentPointer().StringifyUriFragment(sb);
+            invalidDocumentPointer = sb.GetString();
+            sb.Clear();
+            validator.GetInvalidSchemaPointer().StringifyUriFragment(sb);
+            invalidSchemaPointer = sb.GetString();
+
+            // Input JSON is invalid according to the schema
+            OsConfigLogError(ModulesManagerLog::Get(), "Payload JSON at '%s' is invalid according to the schema '%s/%s'", invalidDocumentPointer.c_str(), invalidSchemaPointer.c_str(), validator.GetInvalidSchemaKeyword());
```

Comment:
>at '%s' [](http://example.com/codeflow?start=69&length=7)

How much is logged here? Can actual payload contents be revealed? Can we change to an numeric index? If we keep logging actual payload values, we need to condition this for full logging enabled-only.

Reply:
The invalid document pointer will produce a JSON pointer to the location in the payload that did not align with the schema.

Reply:
Looked into this further and path always points to the JSON root `'#'`. I will remove since it does not add any additional valuable log information and log payload values with full logging only.